src/daemon/instance_settings_handler.cpp

```diff
@@ -153,6 +157,56 @@ void update_disk(const QString& key, const QString& val, mp::VirtualMachine& ins
     }
 }
 
+bool is_bridged(const mp::VMSpecs& spec, const std::string& br_interface)
+{
+    return std::any_of(spec.extra_interfaces.cbegin(),
+                       spec.extra_interfaces.cend(),
+                       [&br_interface](const auto& network) -> bool { return network.id == br_interface; });
+}
+
+void checked_update_bridged(const QString& key,
+                            const QString& val,
+                            mp::VirtualMachine& instance,
+                            mp::VMSpecs& spec,
+                            const std::string& br_interface)
+{
+    auto bridged = mp::BoolSettingSpec{key, "false"}.interpret(val) == "true";
+
+    if (!bridged && is_bridged(spec, br_interface))
+    {
+        throw mp::InvalidSettingException{key, val, "Bridged interface cannot be removed"};
+    }
+
+    if (bridged)
+    {
+        // The empty string in the MAC indicates the daemon that the interface must be configured.
+        spec.extra_interfaces.push_back({br_interface, "", true});
```

Comment:
No, AFAICS, this wouldn't work because `emplace_back` won't use aggregate initialization. See [this](https://stackoverflow.com/questions/13812703/c11-emplace-back-on-vectorstruct) for an example.

Reply:
Ahh that's right, I remember facing that too. Thanks for pointing it out.